edx_api_doc_tools/tests/test_doc_tools.py

```diff
@@ -0,0 +1,64 @@
+"""
+Tests for API docs tooling.
+
+TODO: Possibly make these more comprehensive -- right now, it's basically 'screenshot testing'.
+"""
+from __future__ import absolute_import, unicode_literals
+
+import json
+from os.path import dirname, join
+
+from django.test import SimpleTestCase
+from django.test.utils import override_settings
+
+from ..example import urls as example_urls
+
+
+@override_settings(ROOT_URLCONF=example_urls.__name__)
+class DocViewTests(SimpleTestCase):
+    """
+    Test that the API docs generated from the example Hedgehog API look right.
+
+    How this test works:
+    * In setUpClass, we load example_schema.json
+    * In test_get_data_view, we generate a Swagger schema, and then dump it to
+      actual_schema.json. Then, we make sure actual_schema.json and example_schema.json
+      are equivalent.
+    * In test_get_data_view, we just make sure the browsable docs load with a 200.
+
+    If you make a change to the doc tools or the example API that change the
+    generated schema, here's how to update to update `test_get_data_view`:
+    * Run the test, which should fail with an AssertionError about the
+      generated schema.
+    * Copy the contents of actual_schema.json into https://editor.swagger.io
+    * Make sure the browsable API looks correct,
+      and make sure the contents of the schema file look sane.
+    * If everything looks right, copy actual_schema.json into expected_schema.json
+      and commit the change, making it the new reference schema.
+    """
+    base_path = dirname(__file__)
+    path_of_expected_schema = join(base_path, 'expected_schema.json')
+    path_of_actual_schema = join(base_path, 'actual_schema.json')
+
+    @classmethod
+    def setUpClass(cls):
+        super(DocViewTests, cls).setUpClass()
+        with open(cls.path_of_expected_schema, 'r') as schema_file:
+            # pylint: disable=attribute-defined-outside-init
+            cls.expected_schema = json.load(schema_file)
+
+    def test_get_data_view(self):
+        response = self.client.get('/swagger.json')
+        assert response.status_code == 200
+        actual_schema = response.json()
+        with open(self.path_of_actual_schema, 'w') as f:
+            json.dump(actual_schema, f, indent=4)
+        assert actual_schema == self.expected_schema, (
+            "Generated schema (dumped to actual_schema.json) "
+            "did not match schema loaded from expected_schema.json."
+        )
+
+    def test_get_ui_view(self):
+        response = self.client.get('/api-docs/')
+        assert response.status_code == 200
+        assert 'edX Hedgehog Service API' in response.content.decode('utf-8')
```

Comment:
You mention wanting to beef this up: I think asserting about a handful of other strings will be good.

Reply:
@nedbat Realizing now that this is the maximal thing I can assert, and it's passing because that string is actually the `<title>`. The API spec is loaded via an AJAX call, which populates all page content.